neon_local/src/main.rs

```diff
@@ -896,20 +896,23 @@ fn handle_safekeeper(sub_match: &ArgMatches, env: &local_env::LocalEnv) -> Resul
     Ok(())
 }
 
-fn handle_start_all(sub_match: &ArgMatches, env: &local_env::LocalEnv) -> Result<()> {
+fn handle_start_all(sub_match: &ArgMatches, env: &local_env::LocalEnv) -> anyhow::Result<()> {
+    etcd::start_etcd_process(env)?;
```

Comment:
In current coding etcd is started by both python code and neon_local, we clearly need one of them. The question is which one. Python code gives slightly more handy manual control, but leaving neon_local untested is not nice. I think we can do everything from neon_local, but make it more functional:

- on stop wait handle, bail out if stop failed;
- wait for start by checking status;
- make the whole etcd managing business optional (it might be running elsewhere) -- e.g. bool flag in conf;
- probably add separate etcd start/stop commands. For example, currently ZenithEnvBuilder on __exit__ doesn't call `neon stop`, it shuts down services individually. etcd start/stop can also be useful for some tests.

Or just sit with python version for some more time.

Reply:
I've decided to sit with the python for now without extra big changes.
More on the topic: https://github.com/neondatabase/neon/pull/1648#issuecomment-1125948912